itest/rust/src/object_test.rs

```diff
@@ -416,6 +417,18 @@ fn object_engine_downcast() {
     node3d.free();
 }
 
+#[itest]
+fn object_reject_invalid_downcast() {
+    let pos = Vector3::new(1.0, 2.0, 3.0);
+    let mut node3d: Gd<Node3D> = Node3D::new_alloc();
+    node3d.set_position(pos);
+
+    let object = node3d.share().upcast::<Object>();
+    assert!(object.try_cast::<Node2D>().is_none());
```

Comment:
Doesn't this already fail without this bug fix? From what i remember, #158 only happens when using custom classes, so you'd need to make one/two custom class(es) here to test this downcasting issue properly right?